src/metabase/util/embed.clj

```diff
@@ -98,7 +98,9 @@
                   {:leeway 60})
       ;; if `jwt/unsign` throws an Exception rethrow it in a format that's friendlier to our API
       (catch Throwable e
-        (throw (ex-info (.getMessage e) {:status-code 400}))))))
+        (if (str/starts-with? (ex-message e) "Token is expired")
+          (throw (ex-info "Token is expired" {:status-code 403}))
```

Comment:
Since we'll act on this message on the FE I wanted to make it more explicit.
Initially I thought to add an error code but I stopped at this step for now

Reply:
I think error codes really are the right way to do this here. "Token is expired" really should be i18n'ed, and having the FE hardcode a specific error message and do text matching on it is a recipe for bugs in the future... it's way too easy for someone to not know this specific error message is looked for by the FE and change it or i18n it and then accidentally break stuff.

The FE can look for the 403 status code, right? When you say error codes aren't supported on `embed` endpoints, keep in mind that we can always tweak the middleware or whatever we need to make them supported -- let's take the extra time to do a robust solution rather that makes it hard to shoot yourself in the foot in the future

Reply:
btw all this stuff you wrote here in the PR comment is really useful information, it would be 100x more useful tho if you added it to the code itself as a comment. People really need to know all of this context.

Reply:
Thanks Cam, I looked into it again, it seems that endpoint is supposed to return error codes in some situations but I wasn't able to do it. I [asked for help](https://metaboat.slack.com/archives/C0641E4PB9B/p1706273158446959) on slack and added more context there